client/protocols/export.py

```diff
@@ -0,0 +1,125 @@
+from client.protocols.common import models
+from client.utils import auth
+from client.utils import network
+from datetime import datetime
+from urllib import error
+import client
+import logging
+import os
+import pickle
+
+log = logging.getLogger(__name__)
+
+GAE_DATETIME_FORMAT = "%Y-%m-%d %H:%M:%S.%f"
+
+class ExportProtocol(models.Protocol):
+    """Downloads submissions from the server"""
+
+    def run(self, messages):
+        """If the --export parameter was used, downloads all submissions 
+        for the current assignment from the server and then exits
+        """
+        data = None
+        try:
+            data = pickle.load(open("export_cache.pkl", "rb"))
+            self.access_token = auth.authenticate(self.args.authenticate)
```

Comment:
This authentication should probably happen before the `try/except` block; it seems like you want it to happen regardless of whether or not there is an error.